src/test/java/com/endava/soa_5/context/ScenarioContext.java

```diff
@@ -0,0 +1,32 @@
+package com.endava.soa_5.context;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ScenarioContext {
+    private final Map<Keys, Object> context;
+    private static ScenarioContext scenarioContext;
+
+    private ScenarioContext() {
+        context = new HashMap<Keys, Object>();
+    }
+
+    public static ScenarioContext getScenarioContext() {
+        if (scenarioContext == null) {
+            scenarioContext = new ScenarioContext();
+        }
+        return scenarioContext;
+    }
+
+    public void saveData(Keys key, Object value) {
+        context.put(key, value);
+    }
+
+    public Object getData(Keys key) {
+        return context.get(key);
```

Comment:
I would suggest to add a Class clazz as input parameter and return clazz.cast(context.get(key)). 
To avoid casting each time this object into specific class.